File: pset6/readability/readability.py
from cs50 import get_string

# Get input from the user
text = get_string("Text: ").lower()

# Letters count can be done be verifying all the alpha chars in text
letters = sum(l.isalpha() for l in text)

# Words count can be done be the function split
words = len(text.split())

# We can only count the sentences individually because not all punctuations define a sentence
# Like "," and "'"
sentences = 0
i = 0

for i in range(len(text)):

    # Look for sentences indicator to count the sentences
    if text[i] == "." or text[i] == "?" or text[i] == "!":
        sentences += 1

L = 100 * float(letters / words)
S = 100 * float(sentences / words)
index = round(0.0588 * L - 0.296 * S - 15.8)

# print(f"{letters} {words} {sentences}")

if index < 1:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")

else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Get input from the user
    string text = get_string("Text: ");
    // Initiate letters variable
    int letters = 0;
    // Initiate words variable
    int words = 0;
    // Initiate sentences variable
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        // Verify every char in the text and look for letters
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        // Verify for char ' ' between words to count words
        // Do not count double space as an extra word
        else if (text[i] == ' ' && text[i + 1] != ' ')
        {
            words++;
        }
        // Look for indicators of sentences
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    // Add an extra word because there is no ' ' after the last word
    words ++;
    float L = 100 * (float)letters / words;
    float S = 100 * (float)sentences / words;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)round(index));
    }
}